mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -534,8 +588,13 @@ class Word2VecModel private[spark] (
     // Need not divide with the norm of the given vector since it is constant.
     val cosVec = cosineVec.map(_.toDouble)
     var ind = 0
+    var vecNorm = 1f
+    if (norm)
+      vecNorm = blas.snrm2(vectorSize, fVector, 1)
```

Comment:
@ygcao this is actually a good catch but I believe is covered by [SPARK-7617](https://issues.apache.org/jira/browse/SPARK-7617) and #6245.